OpenRA.Mods.Common/Traits/Buildings/BuildingUtils.cs

```diff
@@ -17,15 +17,24 @@ namespace OpenRA.Mods.Common.Traits
 {
 	public static class BuildingUtils
 	{
-		public static bool IsCellBuildable(this World world, CPos cell, BuildingInfo bi, Actor toIgnore = null)
+		public static bool IsCellBuildable(this World world, CPos cell, ActorInfo ai, BuildingInfo bi, Actor toIgnore = null)
 		{
 			if (!world.Map.Contains(cell))
 				return false;
 
-			if (world.WorldActor.Trait<BuildingInfluence>().GetBuildingAt(cell) != null)
-				return false;
-
-			if (!bi.AllowInvalidPlacement && world.ActorMap.GetActorsAt(cell).Any(a => a != toIgnore))
+			var building = world.WorldActor.Trait<BuildingInfluence>().GetBuildingAt(cell);
+			if (building != null)
+			{
+				if (ai == null)
+					return false;
+
+				var replacementInfo = ai.TraitInfoOrDefault<ReplacementInfo>();
+				if (replacementInfo == null)
+					return false;
+				if (!building.TraitsImplementing<Replaceable>().Any(p => p.AcceptsReplacement(building, replacementInfo.Type)))
```

Comment:
Most changes made.

ToDo's: Have to get rid of AcceptsReplacement in Replaceable.cs as @pchote asked, but have some issues with Overlaps, not working.
```
building.TraitsImplementing<Replaceable>().Any(p => !p.IsTraitDisabled && 
	p.Info.Replacements.Overlaps(replacementInfo.Type));
```
and adding ConditionalTrait wich i do not know what do add and what not. Any advice as alsways welcome : )